src/setup_payload/ManualSetupPayloadParser.h

```diff
@@ -41,7 +42,15 @@ class ManualSetupPayloadParser
     std::string mDecimalStringRepresentation;
 
 public:
-    ManualSetupPayloadParser(std::string decimalRepresentation) : mDecimalStringRepresentation(std::move(decimalRepresentation)) {}
```

Comment:
See TestManualCode.cpp:

```
void TestPayloadParser_PartialPayload(nlTestSuite * inSuite, void * inContext)
{
    CHIP_ERROR err;
    SetupPayload payload;
    std::string decimalString;

    decimalString = "2361087535";
    decimalString += Verhoeff10::ComputeCheckChar(decimalString.c_str());
    NL_TEST_ASSERT(inSuite, decimalString.length() == 11);
    err = ManualSetupPayloadParser(decimalString).populatePayload(payload);
    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
```

I think `Verhoeff10` needs to be updated, and also you could add dashes to some of the cases in `TestPayloadParser_*`

Reply:
> I think Verhoeff10 needs to be updated,

To compute check chars for input strings with dashes?   I would think the way this would normally work is that a payload code is generated, complete with check char, and then dashes are inserted for display.  And on the way back in, the dashes are stripped, then any Verhoeff10 work happens....

Reply:
OK, then can you find a good spot to unit test this and add to `TestManualCode.cpp` ?

Reply:
Yes, I can absolutely do that.